### PR TITLE
pdb

### PR DESCRIPTION
**Pull request by sabel83**

### CODE REVIEW

**Comment by r0mai** [comment]
I tried to run the tests locally, there was one which was failing in `metashell_pp_system_test_wave`:

```
[ RUN      ] preprocessed_echo.echoing_include
/Users/r0mai/workspace/metashell/test/system/app/pp/test_preprocessed_echo.cpp:68: Failure
      Expected: cpp_code(_)
      Which is: {"type":"cpp_code","code":null}
To be equal to: mi.command("#include <string>").front()
      Which is: {"type":"error","msg":"/Users/r0mai/workspace/metashell/bin/app/include/metashell/libcxx/__config:864:1: fatal error: encountered #error directive or #pragma wave stop(): Libc++ requires a feature complete C++11 compiler in C++11 or greater."}
[  FAILED  ] preprocessed_echo.echoing_include (162 ms)
```

I'll take a look at it later, just throwing it out here.

File: lib/core/metaprogram_tracer_clang.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    {
+      const data::metaprogram result = create_metaprogram_from_protobuf_stream(
+          protobuf_stream, mode_,
+          expression_ ? *expression_ : data::cpp_code("<environment>"),
+          data::file_location{}, // TODO something sensible here?
+          evaluation_result);
+      if (result.is_empty() && evaluation_result.is_error())
+      {
+        // Most errors will cause templight to generate an empty trace
+        // We're only intrested in non-empty traces
```
Comment (r0mai) [inline lib/core/metaprogram_tracer_clang.cpp]: I know it's not new code, but typo :)

**Comment by r0mai** [comment]
Nevermind the test failure. It happens on master too on my computer.

**Comment by sabel83** [comment]
I have delayed writing documentation in case major changes come up during the review.

**Pull request merged**